scalanativelib/src/ScalaNativeModule.scala

```diff
@@ -89,18 +92,20 @@ trait ScalaNativeModule extends ScalaModule { outer =>
   def nativeWorkdir = T{ T.dest }
 
   // Location of the clang compiler
-  def nativeClang = T{ scalaNativeWorker().discoverClang }
+  def nativeClang = T{ os.Path(scalaNativeWorker().discoverClang) }
```

Comment:
Do we changed return type here? Can we also make it explicit on the left side? Note an issue in this PR, but as we already touch that line, we should improve it right away.

Reply:
We changed the return type of `discoverClang` to the Java file type, so I needed to convert it back to `os.Path` here in order to preserve the signature of `nativeClang`. I can add type annotations

Reply:
Yeah, sorry for nitpicking. I had our last conversation about binary compatibility in mind. And inferred return types are one source of accidentally incompatibilities.